### ISSUE TITLE
rustdoc weird formatting

### ISSUE THREAD

**Issue by Sgeo** [rust-lang/rust#28647]
Rustdoc does something weird to inline code segments

I tried this code:

```rust
/// This function, `test`, is a test.
pub fn test() {}
```

I expected to see this happen: (Taken on Rust 1.0)
![image](https://cloud.githubusercontent.com/assets/251161/10093426/52ebb60a-6320-11e5-9199-5c635ec6dbe5.png)

Instead, this happened: (Taken on Rust 1.3)
![image](https://cloud.githubusercontent.com/assets/251161/10093435/6c847566-6320-11e5-8b44-1bc610437f0a.png)

## Meta

`rustc --version --verbose`:

```
rustc 1.3.0 (9a92aaf19 2015-09-15)
binary: rustc
commit-hash: 9a92aaf19a64603b02b4130fe52958cc12488900
commit-date: 2015-09-15
host: x86_64-pc-windows-gnu
release: 1.3.0
```

`rustdoc --version --verbose`

```
rustdoc 1.3.0 (9a92aaf19 2015-09-15)
binary: rustdoc
commit-hash: 9a92aaf19a64603b02b4130fe52958cc12488900
commit-date: 2015-09-15
host: x86_64-pc-windows-gnu
release: 1.3.0
```

**Comment by Eljay** [comment]
Looks like a duplicate of #24746 which is already fixed but didn't make it into 1.3.

**Comment by alexcrichton** [comment]
Thanks @Eljay!
